src/entities/Proposal/utils.ts

```diff
@@ -175,8 +186,32 @@ export function isProposalStatus(value: string | null | undefined): boolean {
   }
 }
 
-export function toProposalStatus(value: string | null | undefined, orElse: () => any): ProposalStatus | any {
-  return isProposalStatus(value) ? (value as ProposalStatus) : orElse()
+function toCustomType<FinalType, OrElse, ValueType>(
+  value: ValueType,
+  isType: (value: ValueType) => boolean,
+  orElse: () => OrElse
+): FinalType | OrElse {
+  return isType(value) ? (value as unknown as FinalType) : orElse()
+}
+
+export function toProposalStatus<OrElse>(value: string | null | undefined, orElse: () => OrElse) {
+  return toCustomType<ProposalStatus, OrElse, typeof value>(value, isProposalStatus, orElse)
+}
+
+export function toCatalystType<OrElse>(value: string | null | undefined, orElse: () => OrElse) {
+  return toCustomType<CatalystType, OrElse, typeof value>(value, isCatalystType, orElse)
+}
+
+export function toProposalType<OrElse>(value: string | null | undefined, orElse: () => OrElse) {
+  return toCustomType<ProposalType, OrElse, typeof value>(value, isProposalType, orElse)
+}
+
+export function toPoiType<OrElse>(value: string | null | undefined, orElse: () => OrElse) {
+  return toCustomType<PoiType, OrElse, typeof value>(value, isPoiType, orElse)
+}
+
+export function toSortingOrder<OrElse>(value: string | null | undefined, orElse: () => OrElse) {
+  return toCustomType<SortingOrder, OrElse, typeof value>(value, isSortingOrder, orElse)
```

Comment:
I don't like the orElse callback. I'd change this to default value. But we can discuss this later and change it in another PR.